### PR TITLE
Lucca-ui - Add range-format in directive daterangepicker

### PR DESCRIPTION
**Pull request by BenjaminLY**
**Directive daterangepicker**

New optional parameter `range-format`. When a date format is given, it will be return by filter `luifFriendlyRange` if start date and end date are set.

### CODE REVIEW

**Comment by BenjaminLY** [comment]
@lucienbertin tu pourra jeter un coup d'oeil stp? :-) merci !

**Comment by lucienbertin** [comment]
@BenjaminLY english plz

https://www.youtube.com/watch?v=_x9lSQ1SFLE

File: js/filters/timeFilters.js [hunk 1]
```diff
@@ -57,7 +59,43 @@
...
+			var fallbackLocale = "en";
+			if (!!trads && !!trads[fallbackLocale] && !!trads[fallbackLocale][key]) {
+				return trads[fallbackLocale][key];
+			}
+			if (!!trads && !!trads[fallbackLocale] && !!trads[fallbackLocale][fallbackKey]) {
+				return trads[fallbackLocale][fallbackKey];
+			}
+			// using range format property when start date and end date are valid
+			if (!!trads && fallbackKey === "other") {
+				return trads;
```
Comment (lucienbertin) [inline js/filters/timeFilters.js]: wot ?

your function signature is notr coherent here

File: js/filters/timeFilters.js [hunk 2]
```diff
@@ -57,7 +59,43 @@
 				other: 'von start(LL) bis end(LL)'
 			}
 		};
-		return function (_block, _excludeEnd, _ampm, _translations) {
+		function getTrad(trads, locale, key, fallbackKey) {
```
Comment (lucienbertin) [inline js/filters/timeFilters.js]: `trads` looks like this

```js
{
  en: { 
   sameDay: 'start(dddd, LL)',
   other: 'lol rofl',
  },
  fr: { ... },
  de: { ... },
}
```

File: ts/date-picker/daterangepicker.directive.ts [hunk 3]
```diff
@@ -72,6 +73,7 @@ module lui.datepicker {
 
 		displayStr: string;
 		displayFormat: string;
+		rangeFormat: any;
```
Comment (lucienbertin) [inline ts/date-picker/daterangepicker.directive.ts]: rangeFormat: string

**Comment by BenjaminLY** [comment]
@lucienbertin is this better now?

**Pull request merged**